doc/user/sql/types/text.md

```diff
@@ -27,7 +27,8 @@ Detail | Info
 
 #### From `text`
 
-You cannot cast `text` to any other type.
+You can cast `text` to [all types](../), provided the string is valid input
```

Comment:
You've got `[cast](../../functions/cast)` elsewhere (with a link) which seems smart! Also, if you are going to change this line anyhow, I wonder if there are words that can distinguish between which types can be cast between, and what happens when you do (I think the "provided the .." clause trips up dynamic v static typing).

Reply:
> I wonder if there are words that can distinguish between which types can be cast between, and what happens when you do (I think the "provided the .." clause trips up dynamic v static typing).

I'm confused. What's incorrect about how it's worded? The cast will fail unless the string is valid input for the destination type.

Reply:
My reading of statically typed stuff is that you *can* e.g. cast from `text` to any other type, but that cast may fail if the string is not valid input. In contrast with a dynamically typed system where the resulting type may depend on whether the cast succeeds or fails.